redash/query_runner/axibase_tsd.py

```diff
@@ -0,0 +1,191 @@
+from io import StringIO
+import json
+import logging
+import sys
+import uuid
+
+import pandas as pd
+import atsd_client
+from atsd_client.exceptions import SQLException
+from atsd_client.services import SQLService, MetricsService
+
+from redash.query_runner import *
+from redash.utils import JSONEncoder
+
+logger = logging.getLogger(__name__)
+
+types_map = {
+    'long': TYPE_INTEGER,
+
+    'bigint': TYPE_INTEGER,
+    'integer': TYPE_INTEGER,
+    'smallint': TYPE_INTEGER,
+
+    'float': TYPE_FLOAT,
+    'double': TYPE_FLOAT,
+    'decimal': TYPE_FLOAT,
+
+    'string': TYPE_STRING,
+    'date': TYPE_DATE,
+    'xsd:dateTimeStamp': TYPE_DATETIME
+}
+
+
+def resolve_redash_type(type_in_atsd):
+    """
+    Retrieve corresponding redash type
+    :param type_in_atsd: `str`
+    :return: redash type constant
+    """
+    if isinstance(type_in_atsd, dict):
+        type_in_redash = types_map.get(type_in_atsd['base'])
+    else:
+        type_in_redash = types_map.get(type_in_atsd)
+    return type_in_redash
+
+
+def generate_rows_and_columns(csv_response):
+    """
+    Prepare rows and columns in redash format from ATSD csv response
+    :param csv_response: `str`
+    :return: prepared rows and columns
+    """
+    meta, data = csv_response.split('\n', 1)
+    meta = meta[1:]
+
+    meta_with_padding = meta + '=' * (4 - len(meta) % 4)
+    meta_decoded = meta_with_padding.decode('base64')
+    meta_json = json.loads(meta_decoded)
+    meta_columns = meta_json['tableSchema']['columns']
+
+    df = pd.read_csv(StringIO(data), sep=',')
```

Comment:
I also noticed that Pandas is a dependency of `atsd_client` -- is it really mandatory?